src/api/java/com/minecolonies/api/research/IGlobalResearch.java

```diff
@@ -0,0 +1,138 @@
+package com.minecolonies.api.research;
+
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.crafting.ItemStorage;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraftforge.items.IItemHandler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * Interface defining how a research is.
+ */
+public interface IGlobalResearch
+{
+    /**
+     * Check if this research can be executed at this moment.
+     * @param uni_level the level of the university.
+     * @return true if so.
+     */
+    boolean canResearch(int uni_level, @NotNull final ILocalResearchTree localTree);
+
+    /**
+     * Check if this research can be displayed in the GUI.
+     * @param uni_level the level of the university.
+     * @return true if so.
+     */
+    boolean canDisplay(int uni_level);
+
+    /**
+     * Load the cost for the research from the configuration file.
+     */
+    void loadCostFromConfig();
+
+    /**
+     * Check whether all resources are available to execute the research.
+     * @param inventory the inventory to check in.
+     * @return true if so
+     */
+    boolean hasEnoughResources(final IItemHandler inventory);
+
+    /**
+     * Get the cost list from the research.
+     * @return the list.
+     */
+    List<ItemStorage> getCostList();
+
+    /**
+     * Start the research.
+     * @param player the player starting it.
+     * @param localResearchTree  the local research tree to store in the colony.
+     */
+    void startResearch(@NotNull final PlayerEntity player, @NotNull final ILocalResearchTree localResearchTree);
+
+    /**
+     * Human readable description of research.
+     * @return the description.
+     */
+    String getDesc();
+
+    /**
+     * Getter of the id of the research.
+     * @return the String id.
+     */
+    String getId();
```

Comment:
I want to avoid resource location where they are not strictly necessary, it's a lot of code and storage overhead and extra comparisons etc needed somewhere where we only expect strings to be added. There is no "different mod require different locations" necessity here.